kbase-extension/static/kbase/js/widgets/appWidgets2/view/sequenceView.js

```diff
@@ -200,75 +156,70 @@ define([
               Set focus on the new input control
         */
 
-        function addNewControl(initialValue) {
+        function addNewControl(initialValue, initialDisplayValue) {
             if (initialValue === undefined) {
                 initialValue = Util.copy(itemSpec.data.defaultValue);
             }
-            return Promise.try(() => {
-                const events = Events.make({ node: container });
-                const control = {
-                    // current native value.
-                    value: initialValue,
-                    // the actual input control (or field wrapper around such)
-                    inputControl: null,
-                    // the actual dome node (used?) to which the input control is attached
-                    node: null,
-                    // the current index - note: used by the inputControl
-                    index: null,
-                };
-                const index = viewModel.pushItem(['items'], control);
-                control.index = index;
-                const parent = ui.getElement('control-container');
-                const controlNode = document.createElement('div');
-                parent.appendChild(controlNode);
-                return makeSingleViewControl(control, events)
-                    .then((inputControl) => {
-                        // This adds the control wrapper html.
-                        controlNode.innerHTML = inputControl.content;
-                        // Each wrapper has a node inside with id "id" for
-                        // the control to attach to.
-                        const attachmentNode = document.getElementById(inputControl.id);
-                        control.node = controlNode;
-                        control.inputControl = inputControl;
+            const control = {
+                // current native value.
+                value: initialValue,
+                displayValue: initialDisplayValue,
+                // the actual input control (or field wrapper around such)
+                inputControl: null,
+                // the actual dome node (used?) to which the input control is attached
+                node: null,
+                // the current index - note: used by the inputControl
+                index: null,
+            };
+            const index = viewModel.pushItem(['items'], control);
+            control.index = index;
+            const parent = ui.getElement('control-container');
+            const controlNode = document.createElement('div');
+            parent.appendChild(controlNode);
+            return makeSingleViewControl(control)
```

Comment:
`events` never gets used - remove it from the calls and functions and from the import.